### PR TITLE
Fix for doc issue https://github.com/grafana/grafana/issues/32890

### PR DESCRIPTION
**Pull request by achatterjee-grafana**
Updated the [Security](https://grafana.com/docs/grafana/latest/administration/security/) topic based on @simonc6372's suggestions.

### CODE REVIEW

File: docs/sources/administration/security.md [hunk 1]
```diff
@@ -36,3 +36,11 @@ To address this vulnerability, you can restrict data source query access in the
 
 - Create multiple data sources with some restrictions added in data source configuration that restrict access (like database name or credentials). Then use the [Data Source Permissions]({{< relref "../permissions/datasource_permissions.md" >}}) Enterprise feature to restrict user access to the data source in Grafana.
 - Create a separate Grafana organization, and in that organization, create a separate data source. Make sure the data source has some option/user/credentials setting that limits access to a subset of the data. Not all data sources have an option to limit access.
+
+## Implications of enabling `Anonymous` access to dashboards
+
+When you enable `Anonymous` access to a dashboard, it is publicly available. This section lists the security implications of enabling `Anonymous` access.
```
Comment (oddlittlebird) [inline docs/sources/administration/security.md:42]: Why is "Anonymous" in code format?

File: docs/sources/administration/security.md [hunk 2]
```diff
@@ -36,3 +36,11 @@ To address this vulnerability, you can restrict data source query access in the
...
+
+- Anyone with the URL can access the dashboard. 
+- Anyone can make view calls to the API and list all folders, dashboards and data sources.
```
Comment (oddlittlebird) [inline docs/sources/administration/security.md:45]:
```suggestion
- Anyone can make view calls to the API and list all folders, dashboards, and data sources.
```

File: docs/sources/auth/grafana.md [hunk 3]
```diff
@@ -52,7 +52,7 @@ api_key_max_seconds_to_live = -1
 
 ### Anonymous authentication
 
-You can make Grafana accessible without any login required by enabling anonymous access in the configuration file.
+You can make Grafana accessible without any login required by enabling anonymous access in the configuration file. For more information, refer to [Implications of allowing `Anonymous` access to dashboards]({{< relref "../administration/security.md" >}})
```
Comment (oddlittlebird) [inline docs/sources/auth/grafana.md:55]:
```suggestion
You can make Grafana accessible without any login required by enabling anonymous access in the configuration file. For more information, refer to [Implications of allowing `Anonymous` access to dashboards]({{< relref "../administration/security.md" >}}).
```

**Review by oddlittlebird — APPROVED** [review]
A couple edits, then g2g

File: docs/sources/administration/security.md (hunk 1 above)
Comment (achatterjee-grafana) [reply docs/sources/administration/security.md:42]: I can change it, if it should not be.

Comment (achatterjee-grafana) [reply docs/sources/administration/security.md]: Fixed.

**Pull request merged**